content/blog/learn-pulumi-inputs-outputs/index.md

```diff
@@ -0,0 +1,158 @@
+---
+title: "Learn Pulumi: Inputs and Outputs"
+date: 2020-09-11T08:29:28-05:00
+meta_desc: "Learn how inputs and outputs are key to building infrastructure with code."
+meta_image: learn_pulumi.png
+authors:
+    - sophia-parafina
+tags:
+    - infrastructure as code
+    - Promises
+---
+
+This article is about resource inputs and outputs, what they are, why we have them, and how to use them.
+
+<!--more-->
+
+When we create a resource, you pass arguments to its constructor. They are called the inputs and can be
+
+- concrete values (like strings, numbers, and booleans), arrays, or other structures
+- asynchronous values such as promises or
+- outputs from other resources
+
+After a resource is created, we can access any of its properties. These are called its outputs. Outputs are resolved to values once they're known. Pulumi lets you see how resources change with [preview]({{< relref "docs/reference/cli/pulumi_preview" >}}). A preview shows the difference from the current state to the desired state. The change hasn’t been made, so some outputs might not be known at runtime.
+
+For example, we have a load balancer. It takes a target in a port as inputs, and its outputs include an IP address and a port.
+
+```typescript
+interface LoadBalancerArgs {
+  target: Input<string>;
+  Port: Input<number>;
+}
+
+class LoadBalancer {
```

Comment:
I've reframed this as a pseudo code class for a load balancer and reworked the explanation.

Reply:
Why a pseudo-class and not just actual code? I'm not understanding how it's better to use code that's abstract versus code that's more like what's in, say, most of our examples (since that'll be much more common and actually map to the code you'll actually write).